### ISSUE TITLE
Failing to reconstruct high value pixels with pix2pix

### ISSUE THREAD

**Issue by adriacabeza** [junyanz/pytorch-CycleGAN-and-pix2pix#1059]
Hi, 

Using pix2pix I've experienced bad reconstructions related to colour consistency, especially with really high-valued pixels (white mainly that becomes grey). Firstly I thought it could be the way I was converting the numpy array (which ranges from [-1,1]) wrongly to 0-255:

```python
prediction = (0.5 * prediction + 0.5) * 255.0
#prediction = ((prediction + 1) / 2.0) * 255.0 
```
But it is not. Here an example → [https://imgur.com/gallery/O7hhgXN](https://imgur.com/gallery/O7hhgXN)

What could it be? How can I solve it? Thanks :D

**Comment by junyanz** [comment]
You can increase the lambda for L1 loss (`--lambda_L1`). If you know the mask, you can use the mask and ask the generator to produce pixels inside the mask: `G(x)*m+x*(1-m)`.

**Comment by junyanz** [comment]
Doing it during both training and inference if the mask is given as part of training data.

**Comment by adriacabeza** [comment]
Hi, I've tried to increase L1 lambda loss (150 and 200) and no improvement was experienced... I had the same greyish colors when it should be white. Which range were you thinking in terms of increase for the L1 lambda?

PS: I do not have the mask, so I cannot use it :(

**Comment by junyanz** [comment]
Increasing lambda L1 loss will not change the range of the output. I am not sure how to solve your issue. Maybe try a baseline with L1 only and no GAN loss.

**Comment by adriacabeza** [comment]
Wouldn't that be just like a normal CAE with a L1 loss? I'll try it out :D
